test/functional/page_objects/time_picker.ts

```diff
@@ -125,7 +125,28 @@ export class TimePickerPageObject extends FtrService {
    * @param {String} toTime MMM D, YYYY @ HH:mm:ss.SSS
    */
   public async setAbsoluteRange(fromTime: string, toTime: string) {
-    this.log.debug(`Setting absolute range to ${fromTime} to ${toTime}`);
+    // setAbsoluteRange used to always set the timepicker.  This version checks if the
+    // required times are already in the URL and, if so, returns to save testing time.
+    const currentUrl = await this.browser.getCurrentUrl();
+    const DEFAULT_DATE_FORMAT = 'MMM D, YYYY @ HH:mm:ss.SSS';
+    const startMoment = moment.utc(fromTime, DEFAULT_DATE_FORMAT).toISOString();
+    const endMoment = moment.utc(toTime, DEFAULT_DATE_FORMAT).toISOString();
+    if (currentUrl.includes(startMoment) && currentUrl.includes(endMoment)) {
+      this.log.debug(
+        `We already have the desired start (${fromTime}) and end (${toTime}) in the URL, returning from setAbsoluteRange`
+      );
+      return;
+    } else {
+      this.log.debug(`Setting absolute range to ${fromTime} to ${toTime}`);
+      return await this.setAbsoluteRangeAlways(fromTime, toTime);
+    }
+  }
+
+  /**
+   * @param {String} fromTime MMM D, YYYY @ HH:mm:ss.SSS
+   * @param {String} toTime MMM D, YYYY @ HH:mm:ss.SSS
+   */
+  public async setAbsoluteRangeAlways(fromTime: string, toTime: string) {
```

Comment:
I'm not in love with the name `setAbsoluteTimeAlways` and thought `setAbsoluteTimeForce` might be better?

Reply:
Possible alternative

```
  /**
   * @param {String} fromTime MMM D, YYYY @ HH:mm:ss.SSS
   * @param {String} toTime MMM D, YYYY @ HH:mm:ss.SSS
   * @param {Boolean} force time picker force update, default is false
   */
  public async setAbsoluteRange(fromTime: string, toTime: string, force = false) {
````

Maybe It won't be bad to call function few time like 
`await PageObjects.timePicker.setAbsoluteRange(fromTime, toTime, true);`